File: Karan-Example-Project-Solutions/Python/Numerical/EToNDigits.py
import math

def eToNDigits(n):
    if (n == 0):
        return 0
    return float(str(math.e)[0:n+1])

user = input("How many digits? ")
n = int(user)
if (n > 21):
    print("math.e is only accurate up to 21 digits")
    exit()
print("E to", n, "digits is", eToNDigits(n))

File: Karan-Example-Project-Solutions/Java/Numerical/EToNDigits.java
import java.util.Scanner;
import java.lang.Math;

public class EToNDigits {

  public static void main(String[] args) {
    Scanner input = new Scanner(System.in);
    System.out.println("How many digits? ");
    int n = input.nextInt();

    if (n > 16) {
      System.out.println("Java.lang.Math.E is only accurate up to 16 digits");
      System.exit(0);
    }
    System.out.println("E to " + n + " digits is " + eToNDigits(n));
  }

  private static double eToNDigits(int n) {
    if (n == 0) { return 0; }
    return Double.parseDouble(Double.toString(Math.E).substring(0, n + 1));
  }
}
